aten/src/ATen/native/cuda/AveragePool2d.cu

```diff
@@ -289,13 +281,14 @@ void avg_pool2d_out_cuda_template(
     inputHeight, inputWidth,
     outputHeight, outputWidth);
 
+  const auto memory_format = input_.suggest_memory_format();
```

Comment:
you could make memory_format an input to shape_check, to avoid calling it twice.